File: Deepspace-Java/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author inaki
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{
    private final float EXTRAEFFICIENCY=1.2f;
    Dice dice=new Dice();
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
    }
    
    @Override
    public float fire(){
        int size=getWeapons().size();
        if (dice.extraEfficiency()){
            float factor= EXTRAEFFICIENCY;
            for (int i=0; i< size;i++){
                Weapon w=getWeapons().get(i);
                factor*=w.useIt();
            }
            System.out.println("factor "+factor);
            return getAmmoPower()*factor;
        }
        else{
            float fire=super.fire();
            return fire;
        }
    }
    
    public String toString(){
        return "Beta"+ super.toString();
    }
    
}

File: Deepspace-Ruby/lib/BetaPowerEfficientSpaceStation.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'PowerEfficientSpaceStation'
require_relative 'SpaceStation'

module Deepspace
class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
  
  @@EXTRAEFFICIENCY=1.2
  def initialize(station)
    super(station)
  end
  
  def fire
    dice=Dice.new
    if dice.extraEfficiency
      factor=@@EXTRAEFFICIENCY
      for i in 0...weapons.size
        w=weapons.at(i)
        factor*=w.useIt
      end
      factor
    else
      factor=super
      factor
    end
  end
  
 def setLoot(loot)
    super(Loot.new(loot.nSupplies, loot.nWeapons, loot.nShields, loot.nHangars, loot.nMedals, false, false))
  end
  
  def to_s
    salida="Beta"+super
    salida
  end
end
end